src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -35,8 +35,10 @@ public class RoslynAnalyzerProvider(
     public const string CSharpLanguage = "cs";
     public const string VBNetLanguage = "vbnet";
 
-    private const string LegacyServerPropertyPrefix = "sonaranalyzer-";
-    private const string RoslynRepoPrefix = "roslyn.";
+    private const string ServerPropertyFormat = "sonar.{0}.analyzer";
+    private const string RoslynRepoPrefix = "roslyn."; // Used for plugins generated by SonarQube Roslyn SDK, and legacy Sonar Security C# Frontend
+    private const string LegacySecurityFrontEndPropertyPrefix = "sonaranalyzer.security.cs"; // Should not be used and exists for backward compatibility with version <= 9.2
```

Comment:
The comment doesn't sound right on the Security front end. It's only used for deduplication, and it's not really needed for backward compatibility. 

As this constant was not here before, I'd hardcode the value below where it's used.